build/android/generate_xwalk_shared_library.py

```diff
@@ -0,0 +1,251 @@
+  files_to_copy = [
+      # AndroidManifest.xml from template.
+      'AndroidManifest.xml',
+      # Eclipse project properties from template.
+      'project.properties',
+      # Ant build file.
+      'build.xml',
+      # Ant properties file.
+      'ant.properties',
+  ]
+  for f in files_to_copy:
+    source_file = os.path.join(template_dir, f)
+    target_file = os.path.join(out_project_dir, f)
+
+    shutil.copy2(source_file, target_file)
+
+
+def CopyBinaries(out_dir, out_project_dir, src_package):
+  print 'Copying binaries...'
+
+  libs_dir = os.path.join(out_project_dir, 'libs')
+  if not os.path.exists(libs_dir):
+    os.mkdir(libs_dir)
+
+  # Copy jar files to libs.
+  if src_package:
+    libs_to_copy = [
+    ]
+  else:
+    libs_to_copy = [
+        'xwalk_core_library_java_app_part.jar',
+    ]
+
+  for lib in libs_to_copy:
+    source_file = os.path.join(out_dir, 'lib.java', lib)
+    target_file = os.path.join(libs_dir, lib)
+    shutil.copyfile(source_file, target_file)
+
+
+def CopyDirAndPrefixDuplicates(input_dir, output_dir, prefix):
+  """ Copy the files into the output directory. If one file in input_dir folder
+  doesn't exist, copy it directly. If a file exists, copy it and rename the
+  file so that the resources won't be overrided. So all of them could be
+  packaged into the xwalk core library.
+  """
+  for root, _, files in os.walk(input_dir):
+    for f in files:
+      src_file = os.path.join(root, f)
+      relative_path = os.path.relpath(src_file, input_dir)
+      target_file = os.path.join(output_dir, relative_path)
+      target_dir_name = os.path.dirname(target_file)
+      if not os.path.exists(target_dir_name):
+        os.makedirs(target_dir_name)
+      # If the file exists, copy it and rename it with another name to
+      # avoid overwriting the existing one.
+      if os.path.exists(target_file):
+        target_base_name = os.path.basename(target_file)
+        target_base_name = prefix + '_' + target_base_name
+        target_file = os.path.join(target_dir_name, target_base_name)
+      shutil.copyfile(src_file, target_file)
+
+
+def MoveImagesToNonMdpiFolders(res_root):
+  """Move images from drawable-*-mdpi-* folders to drawable-* folders.
+
+  Why? http://crbug.com/289843
+
+  Copied from build/android/gyp/package_resources.py.
+  """
+  for src_dir_name in os.listdir(res_root):
+    src_components = src_dir_name.split('-')
+    if src_components[0] != 'drawable' or 'mdpi' not in src_components:
+      continue
+    src_dir = os.path.join(res_root, src_dir_name)
+    if not os.path.isdir(src_dir):
+      continue
+    dst_components = [c for c in src_components if c != 'mdpi']
+    assert dst_components != src_components
+    dst_dir_name = '-'.join(dst_components)
+    dst_dir = os.path.join(res_root, dst_dir_name)
+    if not os.path.isdir(dst_dir):
+      os.makedirs(dst_dir)
+    for src_file_name in os.listdir(src_dir):
+      if not src_file_name.endswith('.png'):
```

Comment:
The code around here is just copied from generate_xwalk_core_library. It's better to keep compliance with the core library if no specific needs